tests/integration/integration_test.py

```diff
@@ -1269,6 +1269,108 @@ def verify_avro_explicit_read_schema():
     pass
 
 
+def verify_serializer():
+    from base64 import standard_b64encode as b64encode
+    from base64 import standard_b64decode as b64decode
+    from base64 import b16encode, b16decode
+
+    # base64 deserializer for converting message contents back to a string
+    def b64_decoder(topic, data):
+        if data is None:
+            return None
+        return b64decode(data)
+
+    # base16 deserializer for converting message contents back to a string
+    def b16_decoder(topic, data):
+        if data is None:
+            return None
+        return b16decode(data)
+
+    # Test matrix fo use when evaluating alternative kv serialization strategies.
+    combinations = [
+        # Test default serializers, use encode for python 2 and 3 compatibility.
+        dict(key="b64encode".encode('utf-8'), value="b64encode".encode('utf-8'),
+             key_decoder=b64_decoder, value_decoder=b64_decoder),
+        # Test default key_serializer with value_serializer override
+        dict(key="b64encode".encode('utf-8'), value="b16encode".encode('utf-8'),
+             value_encoder=lambda topic, value: b16encode(value),
+             key_decoder=b64_decoder, value_decoder=b16_decoder),
+        # Test key_serializer override with default value_serializer
+        dict(key="b16encode".encode('utf-8'), value="b64encode".encode('utf-8'),
+             key_encoder=lambda topic, key: b16encode(key),
+             key_decoder=b16_decoder, value_decoder=b64_decoder),
+        # Test key_serializer and value_serializer overrides
+        dict(key="b16encode".encode('utf-8'), value="b16encode".encode('utf-8'),
+             key_encoder=lambda topic, key: b16encode(key),
+             value_encoder=lambda topic, value: b16encode(value),
+             key_decoder=b16_decoder, value_decoder=b16_decoder),
+    ]
+
+    # Test Producer serializer
+    def assert_cb(err, msg, conf):
```

Comment:
this name is a bit vague

Reply:
The name or the comment or both?

Reply:
the method name should say what the method does.